phys/module_diag_afwa.F

```diff
@@ -236,28 +306,30 @@ SUBROUTINE afwa_diagnostics_driver (   grid , config_flags     &
 
     ! Calculate precipitable water
     ! ----------------------------
-    nz=kme-kms+1
-    DO i=ims,ime
-      DO j=jms,jme
-        grid % afwa_pwat ( i, j ) = Pwat( nz,                  &
-                                          qvapr(i,kms:kme,j),  &
-                                          qcloud(i,kms:kme,j), &
-                                          dz8w(i,kms:kme,j),   &
-                                          rho(i,kms:kme,j) )
+    IF ( F_QV .AND. F_QC ) THEN
+      nz=kme-kms+1
+      DO i=ims,ime
+        DO j=jms,jme
+          grid % afwa_pwat ( i, j ) = Pwat( nz,                  &
+                                            qvapr(i,kms:kme,j),  &
+                                            qcloud(i,kms:kme,j), &
+                                            dz8w(i,kms:kme,j),   &
+                                            rho(i,kms:kme,j) )
+        ENDDO
       ENDDO
-    ENDDO
+    END IF
 
     ! After each history dump, reset max/min value arrays
     ! ----------------------------------------------------------------------
     IF ( is_after_history_dump ) THEN
-      DO j = jms, jme
-        DO i = ims, ime
-          grid % wspd10max(i,j) = 0.
-          IF ( config_flags % afwa_severe_opt == 1 ) THEN
-          grid % afwa_llws(i,j) = 0.
-          ENDIF
+      IF ( config_flags % afwa_severe_opt == 1 ) THEN
+        DO j = jms, jme
+          DO i = ims, ime
+            grid % wspd10max(i,j) = 0.
+            grid % afwa_llws(i,j) = 0.
+          ENDDO
```

Comment:
@davegill wspd10max needs to be outside afwa_severe_opt, this one needs to be computed all the time.

Reply:
@davegill Or we can move wspd10max to be packaged with afwa_severe_opt in registry.afwa.

Reply:
@weiwangncar 
Wei,
I think that I fixed this. I made the afwa_diag_opt always active if any other AFWA diag is active. So the wspd10max is always available. Let me know if this addresses your concern adequately.

Reply:
@davegill When only afwa_diag_opt is active, then this can still be a problem.

Reply:
@weiwangncar 
Wei,
I think that I have _now_ fixed the case where you were still getting incorrect memory access